### DISCUSSION TITLE
How does the "Load Image Batch" Node work? I cant get it to do what I want.

### DISCUSSION THREAD

**Discussion by ProtoBelisarius** [WASasquatch/was-node-suite-comfyui#55] [Q&A] [answered]
I want to to use the same prompt and upscaling on all textures in a folder. 
Can I use the "Load Image Batch" Node to just specify a folder and then it repeats the chain on all the images in the folder or is that not how it works? 
If it doesnt work this way, is there an alternative way of doing this?

**Comment by WASasquatch** [comment]
You set a folder, set to increment_image, and then set the number on batches on your comfyUI menu, and then run. It will swap images each run going through the list of images found in the folder. Single image works by just selecting the index of the image. See comments made yesterday about this: https://github.com/WASasquatch/was-node-suite-comfyui/issues/54#issuecomment-15191084

I did want it to be totally different but ComfyUI is pretty limited when it comes to the python nodes without customizing ComfyUI itself.

**Reply by ProtoBelisarius** [reply]
Thank you for the quick answer, but I fear I formulated my question badly. 
The screenshot shows what im trying to do. I essentially want to automatically upscale all images in a folder with a specific upscaler model, then run do img2img with low denoise to remove the upscaling artefacts. With my current chain it just stops after the first image. Did I configure it wrong or is that not a functionality of the node, as in is it intended to increase the index manually and this node is just made so the file picker doesnt have to be opened?
 
![batchUpscale](https://user-images.githubusercontent.com/103048431/234044836-d6728c0a-b931-4f4a-8cbd-76ee84af15dd.png)

**Reply by WASasquatch** [reply, marked as answer]
You have to enable batches in ComfyUI menu. There is a checkbox that enables a slider that you can set how many times ComfyUI runs. Why it's hidden as a probably often used feature, I have no clue. 

![image](https://user-images.githubusercontent.com/1151589/234052152-5611bad4-5c78-4d51-9da7-10a1ceab1d23.png)

**Reply by ProtoBelisarius** [reply]
Thank you very much. I thought it would kinda do it in one "queue prompt" run. Entering a higher batch number works.

**Reply by WASasquatch** [reply]
You should be able to just queue up a bunch of prompts too, if it''s just stopping then something is wrong.

**Reply by Humanoidme** [reply]
Hi @ProtoBelisarius were you able to solve this by increasing the batch count ? Was the Load Image batch getting each image sequentially in the target folder?

**Comment by gacela80** [comment]
Hi newbie here made this batch face restore workflow. I copy/pasted from different workflows I found.
I think... it works like a charm but maybe can get better. sharing to see if it helps out anybody and please let me know if I am doing this right or if there is any way to improve.
You have to go to extra options and put the number of images in folder to batch count.
![Batch_Face_Restore_1_workflow](https://github.com/WASasquatch/was-node-suite-comfyui/assets/139797891/d4a0f4a7-8774-43e8-8d5e-9d0e2f0a98b7)

**Reply by WASasquatch** [reply]
I can't see anything "wrong". There are ways to use this stuff that I didn't even imagine that people have proven time and again. This morning someone wanted a random image loaded, which I said was a easy node to make but didn't know of one. But then someone pointed out it was easy enough with just random number and the Load Image Batch node. Something like this:

![image](https://github.com/WASasquatch/was-node-suite-comfyui/assets/1151589/3ef223fd-279b-4032-ad50-d3c3ccf77e40)

So yeah, lots of ways to use this stuff.
